File: app/channels/solution_channel.rb
class SolutionChannel < ApplicationCable::Channel
  def subscribed
    stream_for solution
  end

  def unsubscribed
  end

  def create_submission(submission)
    uuid = SecureRandom.uuid

    SubmissionServices::Create.(uuid, solution, submission["files"])
  end

  private

  def solution
    @solution ||= Solution.find(params[:id])
  end
end

File: app/javascript/channels/solution_channel.js
import consumer from "./consumer"

class SolutionChannel {
  constructor(id, onReceive) {
    this.id = id;
    this.onReceive = onReceive;
  }

  subscribe() {
    this.subscription = consumer.subscriptions.create(
      { channel: "SolutionChannel", id: this.id }
    );

    this.subscription.received = this.onReceive;
  }

  createSubmission(submission) {
    this.subscription.perform("create_submission", submission);
  }
}

export default SolutionChannel;
